File: t6/Circle.cpp
/**
 * Class that defines a circle.
 *
 * @author João Bolsson (jvmarques@inf.ufsm.br)
 * @since 2017, 17 May.
 */

#ifndef CIRCLE
#define CIRCLE

#include "Shape.cpp"

#define PI 3.14;

class Circle : public Shape {

private:
    int radius;
public:

    /**
     * Default construct.
     *
     * @param r Circle radius.
     */
    Circle(int r) : Shape() {
        radius = r;
    }

    /**
     * @return Circle area.
     */
    double getArea() {
        return radius * radius * PI;
    }

};

#endif

File: t6-java/src/main/Circle.java
package main;

/**
 * Class that defines a circle.
 *
 * @author João Bolsson (joaovictorbolsson@gmail.com)
 * @since 2017, 27 May.
 */
public class Circle extends Shape {

    private final int radius;

    private static final double PI = 3.14;

    /**
     * Default construct.
     *
     * @param r Circle radius.
     */
    public Circle(final int r) {
        radius = r;
    }

    /**
     * @return Circle area.
     */
    @Override
    public double getArea() {
        return radius * radius * PI;
    }

    @Override
    public String toString() {
        return "Círculo";
    }

}
